packages/api/amplify_api/example/integration_test/util.dart

```diff
@@ -0,0 +1,255 @@
+
+  Post post = Post(title: title, rating: rating, blog: createdBlog);
+  final createPostReq = await authorizeRequestForUserPools(
+    ModelMutations.create(post),
+  );
+  final createPostRes =
+      await Amplify.API.mutate(request: createPostReq).response;
+  throwIfError(createPostRes);
+  Post? data = createPostRes.data;
+  if (data == null) {
+    throw Exception(
+      'Null response while creating post. Response errors: ${createPostRes.errors}',
+    );
+  }
+
+  return data;
+}
+
+Future<GraphQLRequest<T>> authorizeRequestForUserPools<T>(
+    GraphQLRequest<T> request) async {
+  // TODO(ragingsquirrel3): convert to authorizationType parameter when implemented.
+  final authSession =
+      await Amplify.Auth.fetchAuthSession() as CognitoAuthSession;
+  final accessToken = authSession.userPoolTokens?.accessToken.raw;
+  if (accessToken == null) {
+    throw const AmplifyException(
+      'Could not get access token from cognito.',
+      recoverySuggestion: 'Ensure test user signed in.',
+    );
+  }
+  final headers = {AWSHeaders.authorization: accessToken};
+
+  return GraphQLRequest<T>(
+    document: request.document,
+    variables: request.variables,
+    modelType: request.modelType,
+    decodePath: request.decodePath,
+    headers: headers,
+  );
+}
+
+GraphQLRequest<T> authorizeRequestForApiKey<T>(
+  GraphQLRequest<T> request,
+  String apiKey,
+) {
+  // TODO(ragingsquirrel3): convert to authorizationType parameter when implemented.
+  final headers = {
+    AWSHeaders.authorization: APIAuthorizationType.none.rawValue,
+    'X-Api-Key': apiKey
+  };
+
+  return GraphQLRequest<T>(
+    document: request.document,
+    variables: request.variables,
+    modelType: request.modelType,
+    decodePath: request.decodePath,
+    headers: headers,
+  );
+}
+
+Future<Blog?> deleteBlog(String id) async {
+  final request = await authorizeRequestForUserPools(
+    ModelMutations.deleteById(Blog.classType, id),
+  );
+  final response = await Amplify.API.mutate(request: request).response;
+  throwIfError(response);
+  blogCache.removeWhere((blog) => blog.id == id);
+  return response.data;
+}
+
+Future<void> deletePost(String id) async {
+  final request = await authorizeRequestForUserPools(
+    ModelMutations.deleteById(Post.classType, id),
+  );
+  final response = await Amplify.API.mutate(request: request).response;
+  throwIfError(response);
+  postCache.removeWhere((post) => post.id == id);
+}
+
+Future<void> deleteTestModels() async {
+  await Future.wait(blogCache.map((blog) => deleteBlog(blog.id)));
+  await Future.wait(postCache.map((post) => deletePost(post.id)));
+}
+
+/// Throws if response `.hasErrors` (any GraphQL errors from server).
+void throwIfError(GraphQLResponse response) {
+  if (response.hasErrors || response.data == null) {
+    throw AmplifyException(
+      'GraphQL error while running request: ${response.errors.toString()}',
+    );
+  }
+}
+
+// Wait for subscription established for given request.
```

Comment:
Make these all doc comments so they can be seen in IDE tooling when writing/updating tests.

```suggestion
/// Wait for subscription established for given request.
```